src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/AtomPub10ServiceDocumentFormatter.cs

```diff
@@ -103,15 +103,15 @@ private Task WriteXml(XmlWriter writer)
                 throw new ArgumentNullException(nameof(reader));
             }
 
-            XmlReaderWrapper wrappedReader = XmlReaderWrapper.CreateFromReader(reader);
-            await wrappedReader.MoveToContentAsync();
+            reader = XmlReaderWrapper.CreateFromReader(reader);
+            await reader.MoveToContentAsync();
```

Comment:
@stephentoub, the awaits in this library should use `.ConfigureAwait(false);`, right? (In a separate PR).

Reply:
This is a discussion we should have. The problem with ConfigureAwait is that it really litters your code a LOT as you need to use it for every await in all public methods. If the await'd method doesn't go async, it become a no-op and you've just burnt some unnecessary CPU cycles. Every one of these async methods will have a synchronous method coming back (they exist on full framework). In WCF we took a slightly different approach. The ConfigureAwait is important to prevent deadlocks when someone calls `.Result` on a Task from an async method and they have a single threaded synchronization context, i.e. in a UI app. As WCF has a synchronous version of every async method, we adopted the philosophy that if you have the single threaded sync context constraint and wish to block waiting for the result, then use the synchronous form of the method inside which we ensure the correct continuation behavior. If you are using the async method variant, you are explicitly saying you are going to use it in an asynchronous manner and await without blocking the thread so we don't do anything there. As this will have sync and async variants of all of these methods, I would prefer to do that for performance reasons.

Reply:
> I would prefer to do that for performance reasons

With such an approach, you're then inviting the costs of passing back to that context on every yielding await.

Our stance in corefx is that all awaits should not post back to the sync ctx. This lib was moved to corefx, and it should follow the same rules.

Reply:
Fair enough. I think it would be worth having a separate conversation about making this easier and less error prone. It only takes one missed await to cause an unexpected deadlock which can be timing sensitive and difficult to debug. Saving the existing sync context, resetting it to default and then restoring it at the end of the method in a finally would be less error prone and more performant as it removes all the ConfigureAwait overhead on every single call.